library/public/types.h

```diff
@@ -58,6 +81,84 @@ class F3D_EXPORT ratio_t
   double Value;
 };
 
+/**
+ * Describe a RGB color.
+ */
+class F3D_EXPORT color_t
+{
+public:
+  color_t() = default;
+  color_t(const std::vector<double>& vec)
+  {
+    if (vec.size() != 3)
+    {
+      throw f3d::type_construction_exception("Provided vector does not have the right size");
+    }
+    std::copy_n(vec.begin(), 3, this->RGB.begin());
+  }
+  color_t(double red, double green, double blue)
+  {
+    this->RGB[0] = red;
+    this->RGB[1] = green;
+    this->RGB[2] = blue;
+  }
+  color_t(const std::initializer_list<double>& list)
+  {
+    if (list.size() != 3)
+    {
+      throw f3d::type_construction_exception("Provided list does not have the right size");
+    }
+    std::copy_n(list.begin(), 3, this->RGB.begin());
+  }
+  operator std::vector<double>() const
+  {
+    return std::vector<double>(this->RGB.begin(), this->RGB.end());
+  }
+  bool operator==(const color_t& other) const
+  {
+    return this->RGB == other.RGB;
+  }
+  bool operator!=(const color_t& other) const
+  {
+    return this->RGB != other.RGB;
+  }
+  double operator[](size_t i) const
+  {
+    if (i >= 3)
+    {
+      throw f3d::type_access_exception("Incorrect index");
+    }
+    return this->RGB[i];
+  }
+  double& operator[](size_t i)
+  {
+    if (i >= 3)
+    {
+      throw f3d::type_access_exception("Incorrect index");
+    }
+    return this->RGB[i];
+  }
+  const double* data() const
+  {
+    return this->RGB.data();
+  }
+  double r() const
+  {
+    return this->RGB[0];
+  }
+  double g() const
+  {
+    return this->RGB[1];
+  }
+  double b() const
+  {
+    return this->RGB[2];
+  }
+
+private:
+  std::array<double, 3> RGB;
```

Comment:
Should we initialize it to black, in order to avoid uninitialized usage?
```suggestion
  std::array<double, 3> RGB = {};
```
Alternatively, we can remove the default constructor.

Reply:
neither `double` nor `ratio_t` are initialized by default, it dont see why color_t should be ?

Reply:
It's safer I suppose. But fine.